src/backend/src/routers/whoarewe.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2024-present Puter Technologies Inc.
+ *
+ * This file is part of Puter.
+ *
+ * Puter is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+const { Context } = require("../util/context");
+const config = require("../config");
+const { eggspress } = require("../util/expressutil");
+
+/**
+ * The /whoarewe endpoint provides environment information to the frontend
+ * before authentication, including whether certain operations will require
+ * captcha verification.
+ *
+ * This allows the frontend to know in advance whether to display captcha
+ * elements without needing to make a request and get a rejection first.
+ */
+const WHOAREWE_GET = eggspress('/whoarewe', {
```

Comment:
We already have `/whoarewe` implemented in PuterHomepageService, so adding another endpoint with the same name will break detection of whether or not user signup is disabled. This behavior should be moved to the endpoint in PuterHomepageService.